### ISSUE TITLE
Token refresh 'always?' fails on public dpop client 

### ISSUE THREAD

**Issue by tnandreas** [DuendeSoftware/IdentityServer#1400]
**Which version of Duende IdentityServer are you using?** 6.3.2

**Which version of .NET are you using?** 6.0.21

 (Have also tried Duende 6.3.3 with .NET 7.0.10 with the same result)

**Describe the bug**

When performing a token refresh with a public dpop client it always fails with the error "The DPoP proof token in the refresh token request does not match the original used." (Line 760 in TokenRequestValidator.cs). 

From debugging results it seems like the assignment of "var thumbprint = jkt as string" in the method GetProofKeyThumbprint always sets the thumbprint to null even though the 'jkt' object has a valid value. With e.g "var thumbprint = jkt.ToString()" it works, but it seems like "as string" doesn't transform the object properly resulting in failure on the token exchange.

I didn't find any unit test on this specific case but i hope you could verify this behaviour or point me in the right direction if i'm missing something.

**To Reproduce**

Request tokens with a dpop header included for a client with requireClientSecret = false. With refresh token from the previous call, send a new token request with grant type refresh_token (also with a dpop header).

**Expected behavior**

A returned result of new issued tokens.

**Log output/exception with stacktrace**

```
The DPoP proof token in the refresh token request does not match the original used.
```

**Additional context**

![duendedpopbug](https://github.com/DuendeSoftware/Support/assets/142788668/c02e33d2-926a-4618-8495-cee6a681ca51)

**Comment by brockallen** [comment]
Yea, let's add a test for this scenario, then fix please.

**Comment by josephdecock** [comment]
Should we backport this to the 6.3.x branch? @tnandreas - do you need this right away, or can you wait for the 7.0 release (we intend to publish previews in November)?

**Comment by tnandreas** [comment]
Hey, thanks for the fast replies. We would preferably not wait until then.. So if it's possible to backport this fix, I would be very grateful if you did!

**Comment by brockallen** [comment]
PR has been merged for this, and NuGets have been released for 6.3.4.
